orders/views.py

```diff
@@ -1,37 +1,42 @@
 import json
 
-from django.http     import JsonResponse
-from django.views    import View
-from django.db       import transaction
+from django.http      import JsonResponse
+from django.views     import View
+from django.db        import transaction
+from django.db.models import Q, F
 
-from products.models import Item
-from orders.models   import Order, OrderItem, Location
-from users.models    import Member
-from users.utils     import login
+from products.models  import Item
+from orders.models    import Order, OrderItem, Location, Cart, OrderItemStatus, OrderStatus
+from users.models     import Member
+from users.utils      import login
 
 class OrderView(View):
     @login
     @transaction.atomic
-    def post(self, request, item_id):
-        data = json.loads(request.body)
-        
-        if Order.objects.filter(member_id = request.user.id, status_id = 1).exists():
-            Order.objects.filter(member_id = request.user.id, status_id = 1).delete()
-
-        order = Order.objects.create(member_id = request.user.id, status_id = 1, location_id = None)
+    def post(self, request):
+        item_id = request.GET.get("item_id", None)
+        data    = json.loads(request.body)      
+                
+        if Order.objects.filter(member_id = request.user.id, status_id = OrderStatus.Status.WAITING.value).exists():
+            Order.objects.filter(member_id = request.user.id, status_id = OrderStatus.Status.WAITING.value).delete()
 
-        OrderItem.objects.create(
-            item_id              = item_id,
-            order_id             = order.id,
-            order_item_status_id = 1,
-            quantity             = data['quantity']
+        order = Order.objects.create(member_id = request.user.id, status_id = OrderStatus.Status.WAITING.value, location_id = None)
+        
+        OrderItem.objects.bulk_create(
+            [OrderItem
+            (item_id             = cart.item_id if not item_id else item_id, 
+            order_id             = order.id, 
+            order_item_status_id = OrderItemStatus.ItemStatus.WAITING.value, 
+            quantity             = cart.quantity if not item_id else data['quantity']
+            )
+            for cart in (Cart.objects.filter(member_id = request.user.id) if not item_id else range(1))]
         )
```

Comment:
잘 수정해주셨네요~👍  
Align만 맞춰주세요 ㅎㅎㅎ

```suggestion
        OrderItem.objects.bulk_create(
            [OrderItem(
                item_id              = cart.item_id if not item_id else item_id, 
                order_id             = order.id, 
                quantity             = cart.quantity if not item_id else data['quantity']
                order_item_status_id = OrderItemStatus.ItemStatus.WAITING.value, 
            ) for cart in (Cart.objects.filter(member_id = request.user.id) if not item_id else range(1))]
        )
```

Reply:
[참고] 이번에 수정하라는 얘기는 아니고 다음에는 아래와 같이 수정해보세요.
반드시 카트 테이블에 저장되어 있는 값으로 주문할 필요가 없고, 카트에 담겨져있는 상품의 id와 수량을 직접 리스트로 받아서 저장하도록 만들면 훨씬 로직이 간단해지고 나중에 `디버깅`하기도 쉬워집니다!

```suggestion
        items = data["items"]
        ..
        OrderItem.objects.bulk_create(
            [OrderItem(
                item_id   = item["id"],
                order_id  = order.id, 
                quantity  = item["quantity"]
                status_id = OrderItemStatus.ItemStatus.WAITING.value
            ) for item in items]
```

Reply:
감사합니다~~